src/routes/mod.rs

```diff
@@ -32,7 +32,15 @@ pub fn active_routes() -> Vec<Route> {
 }
 
 pub fn error_catchers() -> Vec<Catcher> {
-    catchers![not_found, panic]
+    catchers![unprocessable_entity, not_found, panic]
+}
+
+#[catch(422)]
+fn unprocessable_entity() -> JsonValue {
+    json!({
+        "status": "error",
+        "reason": "Unprocessable Entity."
```

Comment:
Can we return the processing error, e.g. what field was missing?

Reply:
I changed substantially the PR so maybe another review is required :) Although now we get `ApiErrors` with the bad json line info.